### PR TITLE
Products

### PR DESCRIPTION
**Pull request by Perpy-del**

### CODE REVIEW

File: models/products.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    },
+    deleted: {
+        type: Boolean,
+        default: false
+    },
+    userId: {
+        type: mongoose.Schema.Types.ObjectId,
+        ref: 'User'
+    },
+    revenue: {
```
Comment (hendrixthecoder) [inline models/products.js:26]: why do we have this?

File: models/products.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        type: Number,
+        required: true,
+    },
+    percentage_increase: {
```
Comment (hendrixthecoder) [inline models/products.js:30]: and this too?

File: controllers/products.js [hunk 3]
```diff
@@ -17,15 +17,20 @@ ProductRouter.post(
       return res.status(400).json({ errors: errors.array() });
     }
     const { product_name, unit_price, quantity, user_id } = req.body;
+    const productRevenue = unit_price * quantity
 
     const existingUser = await User.findById(user_id);
```
Comment (hendrixthecoder) [inline controllers/products.js:22]: the auth middleware makes the user available in the user field of the req object. so basically this should be 
```javascript
const user = await User.findById(req.user.id)
```

collecting the data from the endpoint isnt very sharp, anybidy with a valid token can create a transaction in another users name.

File: controllers/products.js [hunk 4]
```diff
@@ -17,15 +17,20 @@ ProductRouter.post(
...
 
+    if (!existingUser) {
```
Comment (hendrixthecoder) [inline controllers/products.js:24]: this would blow the app up, i think you should just throw an error response. like this;
```javascript
if(!existingUser) return res.json({ error: "User not found." })
```

although this edge case would almost never be needed as the authMiddleware doesnt even let the response get to the app.

**Review by hendrixthecoder — CHANGES_REQUESTED** [review]

File: controllers/products.js (hunk 3 above)
Comment (hendrixthecoder) [reply controllers/products.js:22]: also why is the revenue unit_price * quantity? isnt it supposed to be the difference of the selling price and the cost price?

**Review by hendrixthecoder — CHANGES_REQUESTED** [review]

File: .env.example [hunk 5]
```diff
@@ -1,5 +1,5 @@
-PORT=
-MONGODB_URI=
-SECRET=
+PORT=8080
+MONGODB_URI="mongodb://localhost:27017/tranquilpay"
```
Comment (hendrixthecoder) [inline .env.example:2]: this shouldnt have been done, the .env.example is being pushed to github which means exposing our database url.

**Review by hendrixthecoder — CHANGES_REQUESTED** [review]

File: .env.example (hunk 5 above)
Comment (Perpy-del) [reply .env.example]: Lol... My oga that is localhost not the remote one. Also I have an env file that is hidden by gitignore completely different from this. I did not eventually use these variables sef... So I no expose anything.

File: utils/tools.js [hunk 6]
```diff
@@ -28,14 +30,45 @@ const createTransactionUUID = async () => {
...
-    const increaseAmount = newValue - oldValue;
-    const percentageIncrease = (increaseAmount / oldValue) * 100;
-    return percentageIncrease;
+// function percentageIncrease(oldValue, newValue) {
+//     const increaseAmount = newValue - oldValue;
+//     const percentageIncrease = (increaseAmount / oldValue) * 100;
+//     return percentageIncrease;
+// }
+
+async function resetData() {
```
Comment (hendrixthecoder) [inline utils/tools.js:39]: why are we resetting data?

File: app.js [hunk 7]
```diff
@@ -42,7 +42,7 @@ app.use('/api/login', LoginRouter)
 app.use('/api/auth', authMiddleware, AuthRouter)
 app.use('/api/clients', authMiddleware, ClientRouter)
 app.use('/api/invoice', authMiddleware, InvoiceRouter)
-app.use('/api/products', authMiddleware, ProductRouter)
+app.use('/api/products', ProductRouter)
```
Comment (hendrixthecoder) [inline app.js:45]: why did you remove auth from the product router? thats a route that requires authentication.

**Review by hendrixthecoder — APPROVED** [review]
